src/repositories/indexable-hierarchy-repository.php

```diff
@@ -62,7 +83,7 @@ public function add_ancestor( $indexable_id, $ancestor_id, $depth ) {
 			]
 		);
 
-		return $hierarchy->save();
```

Comment:
I think this should not be here. the $hierarchy is of type `Indexable_Hierarchy`

Reply:
Right, which is a `Yoast\WP\Lib\Model` child, just like `Indexable`.

Are you concerned about regressions? Or it's about code quality?

The way I see it, we either:
* keep this as is, and make the `$indexable` parameter annotation in `save_indexable()` [here](https://github.com/Yoast/wordpress-seo/pull/21335/files#diff-f34c8703bd0d7659a31641ca4ff573ab0e8d5fe2c786309f128f589f44cfe394R184) a `Model` one - because indeed as it is now is misleading
* or revert this line to what it was before (`return $hierarchy->save();`) and return early in the `add_ancestor()` function by checking if `should_index_indexable()` is false.

I _think_ I prefer the former because that way we keep things more centralized, but maybe it's not worth the widening of the parameter in `save_indexable()`, what do you think?

Reply:
fair enough, I'll do so! 👍